.github/workflows/checker.yml

```diff
@@ -0,0 +1,114 @@
+name: Checker Script CI
+
+on:
+  pull_request:
+    branches:
+      - main
+      - master
+
+jobs:
+  run-checker:
+    runs-on: ubuntu-latest
+
+    steps:
+      # Step 1: Checkout the code from the repository
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      # Step 2: Cache cargo dependencies to avoid reinstalling every time
+      - name: Cache cargo registry
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-cargo-
+
+      # Step 3: Cache rust toolchain and binaries (including cargo-stylus)
+      - name: Cache rust toolchain
+        uses: actions/cache@v3
+        with:
+          path: ~/.rustup
+          key: ${{ runner.os }}-rustup-${{ hashFiles('**/rust-toolchain.toml') }}
+          restore-keys: |
+            ${{ runner.os }}-rustup-
+
+      # Step 4: Ensure only rustup manages Rust
+      - name: Ensure only rustup manages Rust
```

Comment:
Based on the action run, it seems that we don't need to perform this step. Also `brew` is not needed since this is an Ubuntu image.

Reply:
It's a bit weird Jose but we need this! I ran issues each time while I got error that wasm-unknown-unknow is not installed. After bunch of searches I found out that I need to do this sometimes on Ubuntu to be sure that Rustup is working which is weird. I'll try to find the stackoverflow page about it and share it but this is the only way we can handle rustup issues on Ubuntu which is weird!